src/profile-logic/profile-data.js

```diff
@@ -683,20 +684,20 @@ export function invertCallstack(thread: Thread): Thread {
 
 export function getSampleIndexClosestToTime(
   samples: SamplesTable,
-  time: number
+  time: number,
+  interval: number
 ): IndexIntoSamplesTable {
-  // TODO: This should really use bisect. samples.time is sorted.
-  for (let i: number = 0; i < samples.length; i++) {
-    if (samples.time[i] >= time) {
-      if (i === 0) {
-        return 0;
-      }
-      const distanceToThis = samples.time[i] - time;
-      const distanceToLast = time - samples.time[i - 1];
-      return distanceToThis < distanceToLast ? i : i - 1;
-    }
+  // Bisect to find the index of the first sample after the provided time.
+  const index = bisection.right(samples.time, time);
+
+  if (index === 0) {
+    return 0;
```

Comment:
I think you also need to check if `index === samples.length`, because that's what you'll get when you click after the last sample. And then `samples.time[index]` below would give undefined.